### PR TITLE
New features!

### LINKED ISSUE #52: Get the post id
The post id is useful for having a sort of hash to detect the changes in a post. For instance, if a post changes by the time, we just scrape that one more time. This post id can be get from the `href` attribute of the date `a` element at the bottom of the author's name.  Once the post id is got the `GropPost` class should be updated

### PR DESCRIPTION
**Pull request by iMrDJAi**
What's new?  
Added new fields to the Post interface:  
```js
export default interface Post {
  authorName: string,
  authorUrl: string | null,
  authorAvatar: string | null,
  date: string,
  permalink: string,
  id: string,
  contentText: string | null,
  contentHtml: string | null,
  images: any[]
}
```  
Yup, I've added all of these. It been kinda tricky since I had to make it hover on the post link under the author name to grab the date.

What's missing?  
Videos. I haven't add an implementation for them yet. Also I haven't update the example.

### CODE REVIEW

**Comment by kaanyagci** [comment]
@iMrDJAi looks good! 
I just changed the base branch from `master` to `development`. Do you mind fixing merge conflicts?

**Comment by iMrDJAi** [comment]
All conflicts are fixed!

**Comment by iMrDJAi** [comment]
@kaanyagci Shouldn't we gitignore tsconfig.tsbuildinfo?  
Also I'm using node v14.17.2, and the library worked perfectly, probably you should double check this: https://github.com/Makepad-fr/fbjs/blob/d89e8367039c66a745008e1b2ada7cb0d24d6e47/example/README.md#L5

File: src/lib/models/fb.ts [hunk 1]
```diff
@@ -269,19 +285,23 @@ export default class Facebook {
...
-          outputFileName!,
-          JSON.stringify(allPublicationsList, undefined, 4),
-          { encoding: 'utf8' },
-        );
-      }
+      fs.writeFileSync(
+        outputFileName!,
+        JSON.stringify(allPublicationsList, undefined, 4),
+        { encoding: 'utf8' },
+      );
```
Comment (kaanyagci) [inline src/lib/models/fb.ts:295]: I'm not really sure but I think this change will save all questions even the `save` is false

**Comment by kaanyagci** [comment]
@iMrDJAi actually we can `.gitignore` the tsconfig.tsbuildinfo build info file but the advantage of tracking this file is to decrease the compilation time by incremental compiling.

**Comment by kaanyagci** [comment]
For the `README` lines you're right. The written version is the version that I'm using on my computer, but if you `node` v14.17.2 works great, I'll update this. Do you mind share your `npm` version too?

File: src/lib/models/fb.ts [hunk 2]
```diff
@@ -335,10 +333,8 @@ export default class Facebook {
       );
       if (postHnd?.toString() !== 'JSHandle:undefined') {
         const postData = await this.parsePost(<ElementHandle>postHnd);
-        if (callback !== undefined && callback !== null) {
-          callback(postData);
-        }
-        savePost(postData);
+        if (callback) callback(postData);
+        if (save) savePost(postData);
```
Comment (iMrDJAi) [inline src/lib/models/fb.ts:337]: @kaanyagci I added that check here.

**Comment by iMrDJAi** [comment]
> @iMrDJAi actually we can `.gitignore` the tsconfig.tsbuildinfo build info file but the advantage of tracking this file is to decrease the compilation time by incremental compiling.

Roger that.

**Comment by iMrDJAi** [comment]
> For the `README` lines you're right. The written version is the version that I'm using on my computer, but if you `node` v14.17.2 works great, I'll update this. Do you mind share your `npm` version too?

@kaanyagci  Sure, my npm version is 6.14.13, but this doesn't change the fact that it may work on a lower version, technically you should target the min supported version of puppeteer (node v8.x should work fine, [check this](https://glitch.com/edit/#!/puppeteer-sample?path=package.json%3A14%3A8)).
In the other side I couldn't Install the dev dependencies with node v10 (I used to have this before upgrading to v14), so the min supported version for the development machine is higher than the min version for production.

File: src/lib/models/fb.ts (hunk 2 above)
Comment (kaanyagci) [reply src/lib/models/fb.ts:337]: Yes in git diff I was unable to see the function name but actually you're right.

**Comment by kaanyagci** [comment]
Got it ! Thanks I'll update the README ASAP.

**Review by kaanyagci — APPROVED** [review]

**Comment by iMrDJAi** [comment]
Also don't forget to update the example:  
https://github.com/Makepad-fr/fbjs/blob/d89e8367039c66a745008e1b2ada7cb0d24d6e47/example/groupName123

**Comment by kaanyagci** [comment]
There's still the email issue that you had the last time. I will fix this wile merging though. I've a meeting right know I'll do it just after that

**Comment by iMrDJAi** [comment]
> There's still the email issue that you had the last time. I will fix this wile merging though. I've a meeting right know I'll do it just after that

@kaanyagci  I found out that when I created the previous pull request from the web interface it included my email automatically with the associated commit since it wasn't set to private on my account settings.  
Yeah there is nothing to do about that, any commit I've done on the web interface exposed my email, but anyway this one won't because I changed my account settings.

**Pull request merged**